src/force/CabanaPD_ForceModels_LPS.hpp

```diff
@@ -0,0 +1,146 @@
+/****************************************************************************
+ * Copyright (c) 2022-2023 by Oak Ridge National Laboratory                 *
+ * All rights reserved.                                                     *
+ *                                                                          *
+ * This file is part of CabanaPD. CabanaPD is distributed under a           *
+ * BSD 3-clause license. For the licensing terms see the LICENSE file in    *
+ * the top-level directory.                                                 *
+ *                                                                          *
+ * SPDX-License-Identifier: BSD-3-Clause                                    *
+ ****************************************************************************/
+
+#ifndef FORCE_MODELS_LPS_H
+#define FORCE_MODELS_LPS_H
+
+#include <CabanaPD_ForceModels.hpp>
+#include <CabanaPD_Types.hpp>
+
+namespace CabanaPD
+{
+template <>
+struct ForceModel<LPS, Elastic> : public BaseForceModel
+{
+    using base_type = BaseForceModel;
+    using base_model = LPS;
+    using fracture_type = Elastic;
+
+    using BaseForceModel::delta;
+
+    int influence_type;
+
+    double K;
+    double G;
+    double theta_coeff;
+    double s_coeff;
+
+    ForceModel(){};
+    ForceModel( const double _delta, const double _K, const double _G,
+                const int _influence = 0 )
+        : base_type( _delta )
+        , influence_type( _influence )
+    {
+        set_param( _delta, _K, _G );
+    }
+
+    void set_param( const double _delta, const double _K, const double _G )
+    {
+        delta = _delta;
+        K = _K;
+        G = _G;
+
+        theta_coeff = 3.0 * K - 5.0 * G;
+        s_coeff = 15.0 * G;
+    }
+
+    KOKKOS_INLINE_FUNCTION double influence_function( double xi ) const
+    {
+        if ( influence_type == 1 )
+            return 1.0 / xi;
+        else
+            return 1.0;
+    }
+};
+
+template <>
+struct ForceModel<LPS, Fracture> : public ForceModel<LPS, Elastic>
+{
+    using base_type = ForceModel<LPS, Elastic>;
+    using base_model = LPS;
```

Comment:
@streeve This line is identical to Line 24. Is that the way it should be or should we inherit the base_model from the one from ForceModel<LPS, Elastic>?

Reply:
Not a big deal either way, as long as the tags match